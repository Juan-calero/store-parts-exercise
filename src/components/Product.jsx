import React from "react";
import { string } from "prop-types";
import { Link } from "react-router-dom";
import { item } from "./product.module.scss";

const Product = ({ name, type, price }) => {
	return (
		<li>
			<Link
				to={{
					pathname: `/product/${name}`,
					name: name,
					type: type,
					price: price,
				}}
				className={item}
			>
				<p>{name}</p>
				<p>{type}</p>
				<p>{price}</p>
			</Link>
		</li>
	);
};

Product.propTypes = {
	name: string.isRequired,
	type: string.isRequired,
	price: string.isRequired,
};

export default Product;
