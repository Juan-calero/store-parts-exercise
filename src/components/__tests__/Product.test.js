import { render, screen, cleanup } from "@testing-library/react";
import { BrowserRouter as Router } from "react-router-dom";
import renderer from "react-test-renderer";
import Product from "../Product";

test("Should match snapshot", () => {
	const product = { name: "Mouse 1", price: "112.00$", type: "Mouse" };
	render(
		<Router>
			<Product {...product} />
		</Router>
	);
	const tree = renderer
		.create(
			<Router>
				<Product {...product} />
			</Router>
		)
		.toJSON();
	expect(tree).toMatchSnapshot();
});
