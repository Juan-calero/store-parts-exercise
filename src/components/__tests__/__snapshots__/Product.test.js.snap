// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should match snapshot 1`] = `
<li>
  <a
    className="item"
    href="/product/Mouse 1"
    onClick={[Function]}
  >
    <p>
      Mouse 1
    </p>
    <p>
      Mouse
    </p>
    <p>
      112.00$
    </p>
  </a>
</li>
`;
